File: src/main/java/kitchenpos/menus/tobe/infra/DefaultTobeProductClient.java
package kitchenpos.menus.tobe.infra;


import kitchenpos.menus.tobe.domain.menu.TobeProductClient;
import kitchenpos.products.tobe.domain.TobeProductRepository;
import org.springframework.stereotype.Component;

import java.math.BigDecimal;
import java.util.UUID;

@Component
public class DefaultTobeProductClient implements TobeProductClient {
    private final TobeProductRepository tobeProductRepository;

    public DefaultTobeProductClient(TobeProductRepository tobeProductRepository) {
        this.tobeProductRepository = tobeProductRepository;
    }

    @Override
    public BigDecimal getProductPrice(final UUID productId) {
        return tobeProductRepository.findById(productId).orElseThrow().getPriceValue();
    }
}

File: src/main/java/kitchenpos/menus/tobe/domain/menu/DefaultTobeProductClient.kt
package kitchenpos.menus.tobe.domain.menu

import kitchenpos.products.tobe.domain.TobeProductRepository
import org.springframework.stereotype.Component
import java.util.*

@Component
class DefaultTobeProductClient(
    private val tobeProductRepository: TobeProductRepository,
) : TobeProductClient {
    override fun validateAllProductsExists(menuProductIds: List<UUID>): Boolean {
        val existProductsIds =
            tobeProductRepository.findAllByIdIn(menuProductIds)
                .map { it.id }

        return existProductsIds.containsAll(menuProductIds)
    }
}
